src/libs/actions/Report.js

```diff
@@ -158,6 +158,101 @@ function getSimplifiedReportObject(report) {
     };
 }
 
+/**
+ * Get a simplified version of an IOU report
+ *
+ * @param {Object} reportData
+ * @param {Number} reportData.transactionID
+ * @param {Number} reportData.amount
+ * @param {String} reportData.currency
+ * @param {String} reportData.created
+ * @param {String} reportData.comment
+ * @param {Object[]} reportData.transactionList
+ * @param {String} reportData.ownerEmail
+ * @param {String} reportData.managerEmail
+ * @param {Number} reportData.reportID
+ * @returns {Object}
+ */
+function getSimplifiedIOUReport(reportData) {
+    const transactions = _.map(reportData.transactionList, transaction => ({
+        transactionID: transaction.transactionID,
+        amount: transaction.amount,
+        currency: transaction.currency,
+        created: transaction.created,
+        comment: transaction.comment,
+    }));
+
+    return {
+        reportID: reportData.reportID,
+        ownerEmail: reportData.ownerEmail,
+        managerEmail: reportData.managerEmail,
+        currency: reportData.currency,
+        transactions,
+    };
+}
+
+/**
+ * Fetches the updated data for an IOU Report and updates the IOU collection in Onyx
+ *
+ * @param {Object} report
+ * @param {Object[]} report.reportActionList
+ * @param {Number} report.reportID
+ */
+function updateIOUReportData(report) {
+    const reportActionList = report.reportActionList || [];
+    const containsIOUAction = _.any(reportActionList,
+        reportAction => reportAction.action === CONST.REPORT.REPORT_ACTION_TYPE.IOU);
+
+    // If there aren't any IOU actions, we don't need to fetch any additional data
+    if (!containsIOUAction) {
+        return;
+    }
+
+    // If we don't have one participant (other than the current user), this is not an IOU
+    const participants = getParticipantEmailsFromReport(report);
+    if (participants.length !== 1) {
+        Log.alert('[Report] Report with IOU action has more than 2 participants', true, {
+            reportID: report.reportID,
```

Comment:
Maybe include the participants in this log? I think that would be helpful for troubleshooting it.